community/modules/compute/schedmd-slurm-gcp-v5-partition/variables.tf

```diff
@@ -0,0 +1,298 @@
+  type        = map(any)
+  default     = {}
+}
+
+variable "node_count_dynamic_max" {
+  description = "Maximum number of nodes allowed in this partition"
+  type        = number
+  default     = 10
+}
+
+variable "source_image_project" {
+  type        = string
+  description = "Project where the source image comes from. If it is not provided, the provider project is used."
+  default     = ""
+}
+
+variable "source_image_family" {
+  type        = string
+  description = "Source image family."
+  default     = ""
+}
+
+variable "source_image" {
+  type        = string
+  description = "Source disk image."
+  default     = ""
+}
+
+variable "tags" {
+  type        = list(string)
+  description = "Network tag list."
+  default     = []
+}
+
+variable "disk_type" {
+  description = "Type of boot disk to create for the partition compute nodes"
+  type        = string
+  default     = "pd-standard"
+}
+
+variable "disk_size_gb" {
+  description = "Size of boot disk to create for the partition compute nodes"
+  type        = number
+  default     = 30
+}
+
+variable "disk_auto_delete" {
+  type        = bool
+  description = "Whether or not the boot disk should be auto-deleted."
+  default     = true
+}
+
+variable "additional_disks" {
+  description = "Configurations of additional disks to be included on the partition nodes"
+  type = list(object({
+    disk_name    = string
+    device_name  = string
+    disk_size_gb = number
+    disk_type    = string
+    disk_labels  = map(string)
+    auto_delete  = bool
+    boot         = bool
+  }))
+  default = []
+}
+
+variable "enable_confidential_vm" {
+  type        = bool
+  description = "Enable the Confidential VM configuration. Note: the instance image must support option."
+  default     = false
+}
+
+variable "enable_shielded_vm" {
+  type        = bool
+  description = "Enable the Shielded VM configuration. Note: the instance image must support option."
+  default     = false
+}
+
+variable "enable_oslogin" {
+  type        = bool
+  description = <<-EOD
+    Enables Google Cloud os-login for user login and authentication for VMs.
+    See https://cloud.google.com/compute/docs/oslogin
+    EOD
+  default     = true
+}
+
+variable "can_ip_forward" {
+  description = "Enable IP forwarding, for NAT instances for example"
+  type        = bool
+  default     = false
+}
+
+variable "disable_smt" {
+  type        = bool
+  description = "Disables Simultaneous Multi-Threading (SMT) on instance."
+  default     = false
+}
+
+variable "labels" {
+  description = "Labels to add to partition compute instances. List of key key, value pairs."
+  type        = any
+  default     = {}
+}
+
+variable "min_cpu_platform" {
+  description = "The name of the minimum CPU platform that you want the instance to use."
+  type        = string
+  default     = null
+}
+
+variable "on_host_maintenance" {
+  type        = string
+  description = "Instance availability Policy"
+  default     = "MIGRATE"
```

Comment:
Placement is enabled by default which I believe conflicts with live migration. If this is ignored when placement is true then we should say so in the description.

Reply:
Good point - I don't see this handled anywhere in the submodules. From what I can tell, both on_host_maintenance=MIGRATE and enable_placement=true can be set and neither override at the terraform level. 

I infer from [the docs](https://cloud.google.com/compute/docs/instances/define-instance-placement#support_and_restrictions ) that if on_host_maintenance is not TERMINATE, placement will not be set. Does that seem right?

I'll run some tests to check the behavior and edit the defaults/description accordingly.

Reply:
I was right in my interpretation - I updated the default to TERMINATE so that placement policy will not be overwritten. I also added an extra note in this description to make that behavior clear.